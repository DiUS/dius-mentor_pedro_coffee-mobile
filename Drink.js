'use strict';

import React, { Component } from 'react';
import { View, ListView, Button, Text, StyleSheet } from 'react-native';
import ProceedButton from './ProceedButton';
import hotDrinks from './default_data/hotDrinks.json';
import ExtrasListView from './ExtrasListView';
import OptionsListView from './OptionsListView';
import ApiUtils from './ApiUtils';

//4 Steps detected in this view
var STEP_INITIAL = 0;
var STEP_STYLE = 1;
var STEP_SIZE = 2;
var STEP_EXTRAS = 3;

var ds = new ListView.DataSource({rowHasChanged : (r1, r2) => r1 !== r2});

class Drink extends Component {
  constructor(props) {
    super(props);
    this._isNew = this.props.drink?false:true;
    this.state = {
      order: this.props.order,
      drink: {
        type:null,
        style:this.props.drink?this.props.drink.style:null,
        size:this.props.drink?this.props.drink.size:null
      },
      step: this.props.drink?STEP_EXTRAS:STEP_INITIAL,
      //Visible Options
      drinkOptions: {
        styleOptions:null,
        sizeOptions:null
      },
      //TODO FETCH DATA FROM API && SAME WITH EXTRAS LIST
      optionsList: ds.cloneWithRows(hotDrinks)
    };
  }

  onBack(){
    this.props.onComplete(this.state.order);
  }

  onAddDrink(){
    //TODO PATH WILL CHANGE DEPENDING ON DRINK (IF THERE ARE DIFF DRINKS)
    return fetch(ApiUtils.ENDPOINT+this.state.order.id+'/'+ApiUtils.COFFEE_PATH,{
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        style : this.state.drink.style,
        size : this.state.drink.size,
      })
    })
    .then(ApiUtils.checkStatus)
    .then((response) => response.json())
    .then((responseJson) => {
      var newDrinks = this.state.order.coffees;
      if(!newDrinks){
        newDrinks = [responseJson];
      }
      else{
        newDrinks = [...newDrinks,responseJson];
      }
      var newOrder = this.state.order;
      newOrder.coffees = newDrinks;
      this.props.onComplete(newOrder);
    })
    .catch((error) => {
      //TODO Reset to Initial View
      console.error(error);
    })
  }

  onSelectType(value){//Called on Initial Step. Initial Step is mandatory
    var currentDrink = this.state.drink;
    var currentDrinkOptions = this.state.drinkOptions;
    currentDrink.type=value.title;
    currentDrinkOptions.styleOptions=value.type;
    currentDrinkOptions.sizeOptions=value.size;
    var step=STEP_INITIAL;
    if (value.type)       {step=STEP_STYLE;}
    else if (value.size)  {step=STEP_SIZE;}
    else                  {step=STEP_EXTRAS;}
    this.setState({
      drinkOptions: currentDrinkOptions
    });
    this.onStep(step,currentDrink);
  }

  onSelectStyle(style){
    var currentDrink = this.state.drink;
    currentDrink.style = style.title;
    var step=this.state.step;
    if(this.state.drinkOptions.sizeOptions) {step=STEP_SIZE;}
    else                                    {step=STEP_EXTRAS;}
    this.onStep(step,currentDrink);
  }

  onSelectSize(size){
    var currentDrink = this.state.drink;
    currentDrink.size = size.title;
    this.onStep(STEP_EXTRAS,currentDrink);
  }

  onStep(step,drink){
    var currentDrink = drink?drink:this.state.drink;
    var currentOptions = null;
    switch (step) {
      case STEP_INITIAL: //initial
        currentDrink.type = null;
        currentDrink.style = null;
        currentDrink.size = null;
        currentOptions=ds.cloneWithRows(hotDrinks);
        break;
      case STEP_STYLE: //choose style
        currentDrink.style = null;
        currentDrink.size = null;
        currentOptions=ds.cloneWithRows(this.state.drinkOptions.styleOptions);
        break;
      case STEP_SIZE: //choose size
        currentDrink.size = null;
        currentOptions=ds.cloneWithRows(this.state.drinkOptions.sizeOptions);
        break;
      default:break;
    }
    this.setState({
      drink : currentDrink,
      optionsList : currentOptions,
      step: step
    });
  }

  _renderRowSelect(rowData){
    switch (this.state.step) {
      case STEP_INITIAL:
        return (<Button title={rowData.title} onPress={()=>this.onSelectType(rowData)}/>);
        break;
      case STEP_STYLE:
        return (<Button title={rowData.title} onPress={()=>this.onSelectStyle(rowData)}/>);
        break;
      case STEP_SIZE:
        return (<Button title={rowData.title} onPress={()=>this.onSelectSize(rowData)}/>);
        break;
      default:break;
    }
  }

  _renderSelectedValues(){
    return(
      <View style={{
        flex: 0.45,
        alignItems: 'flex-start',
        alignSelf:'stretch',
      }}>
        <View style={styles.container}>
          {this.state.drink.type &&
            <View style={styles.margin}>
              <Button title={this.state.drink.type} onPress={()=>this.onStep(STEP_INITIAL)} color='#c8dcf4'/>
            </View>
          }
          {this.state.drink.style &&
            <View style={styles.margin}>
             <Button title={this.state.drink.style} onPress={()=>this.onStep(STEP_STYLE)} color='#c8dcf4'/>
            </View>
          }
          {this.state.drink.size &&
            <View style={styles.margin}>
              <Button title={this.state.drink.size} onPress={()=>this.onStep(STEP_SIZE)} color='#c8dcf4'/>
            </View>
          }
        </View>
      </View>
    );
  }

  _renderOptions(){
    if(this.state.step>2){
      return <ExtrasListView/>;
    }
    else{
      return <OptionsListView
        renderRow={this._renderRowSelect.bind(this)}
        dataSource={this.state.optionsList}/>;
    }
  }

  render() {
    return (
      <View style={{flex: 1, padding: 10}}>
        <Text>Current Selection:</Text>
        {this._renderSelectedValues()}
        <Text>Choose your {this.state.step>STEP_SIZE?"extras":"drink"}:</Text>
        {this._renderOptions()}
        <View style={{flexDirection:'row', justifyContent:'space-between'}}>
          <View style={{flex : 1, marginRight:4}}>
            <Button title='<' onPress={()=>this.onBack()} />
          </View>
          <View style={{flex : 8, marginLeft:4}}>
            <ProceedButton title='Add Coffee' onPress={()=>this.onAddDrink()} disabled={!(this.state.step>STEP_SIZE) || !this._isNew}/>
          </View>
        </View>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container:{
    flex:1,
    alignItems:'stretch',
    alignSelf:'stretch',
  },
  margin:{
    marginTop: 8,
  }
});

export default Drink;
