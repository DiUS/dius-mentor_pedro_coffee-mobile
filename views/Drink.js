'use strict';

import React, { Component } from 'react';
import { View, ListView, Button, Text, StyleSheet, ActivityIndicator } from 'react-native';
import Options from '../components/Options';
import hotDrinks from '../default_data/hotDrinks.json';
import ExtrasListView from '../components/ExtrasListView';
import OptionsListView from '../components/OptionsListView';
import { baseUrl } from '../config.js'
import Client from '../client/Client';

const client = Client(baseUrl);

//4 Steps (States) detected in this view
var STEP_INITIAL = 0;
var STEP_STYLE = 1;
var STEP_SIZE = 2;
var STEP_EXTRAS = 3; //This is actually a ghost STEP as is not currently enabled

var ds = new ListView.DataSource({rowHasChanged : (r1, r2) => r1 !== r2});

class Drink extends Component {
  constructor(props) {
    super(props);
    this._initialDrink = this.props.drink;
    this.state = {
      order: this.props.order,
      //TODO: Modify when API accepts other drinks for drink:this.props.drink
      drink: {
        id:this.props.drink?this.props.drink.id:null,
        type:this.props.drink?'Coffee':null, //Only coffee accepted right now
        style:this.props.drink?this.props.drink.style:null,
        size:this.props.drink?this.props.drink.size:null
      },
      isLoading:false,
      step: this.props.drink?STEP_EXTRAS:STEP_INITIAL,
      drinkOptions:null,
      //TODO FETCH DATA FROM API IN componentDidMount
      optionsList: ds.cloneWithRows(hotDrinks)
    };
  }

  componentDidMount(){
    //FETCH MENU WHEN MORE DRINKS AVAILABLE --> NOW HOTDRINKS DEFAULT DATA
  }

  onBack(){
    this.props.onComplete(this.state.order);
  }

  onSelectType(type){//Called on Initial Step. Initial Step is mandatory
    var currentDrink = this.state.drink;
    currentDrink.type=type;
    currentDrink.style=null;
    currentDrink.size=null;
    this.onStep(currentDrink);
  }

  onSelectStyle(style){
    var currentDrink = this.state.drink;
    currentDrink.style = style;
    currentDrink.size=null;
    this.onStep(currentDrink);
  }

  onSelectSize(size){
    var currentDrink = this.state.drink;
    currentDrink.size = size;
    this.onStep(currentDrink);
  }

  onStep(drink){
    if(!this.state.drinkOptions){
      this.setState({
        drink:drink,
        isLoading:this.state.drinkOptions?false:true
      });
      //TODO: By now only coffee options
      this.fetchDrinkOptions('coffee');
    }
    else{
      var currentOptions = null;
      var step=STEP_INITIAL;
      if(!drink.type) {
        step=STEP_INITIAL;
      }
      else if (!drink.style && this.state.drinkOptions.style) {
        step=STEP_STYLE;
      }
      else if (!drink.size && this.state.drinkOptions.size) {
        step=STEP_SIZE;
      }
      else {
        step=STEP_EXTRAS;
      }
      switch (step) {
        case STEP_INITIAL: //initial
          currentOptions=ds.cloneWithRows(hotDrinks);
          break;
        case STEP_STYLE: //choose style
          currentOptions=ds.cloneWithRows(this.state.drinkOptions.style);
          break;
        case STEP_SIZE: //choose size
          currentOptions=ds.cloneWithRows(this.state.drinkOptions.size);
          break;
        default:break;
      }
      this.setState({
        drink: drink,
        optionsList : currentOptions,
        step: step
      });
    }
  }

  renderRowSelect(rowData){
    switch (this.state.step) {
      case STEP_INITIAL:
        return (<Button title={rowData} onPress={()=>this.onSelectType(rowData)}/>);
        break;
      case STEP_STYLE:
        return (<Button title={rowData} onPress={()=>this.onSelectStyle(rowData)}/>);
        break;
      case STEP_SIZE:
        return (<Button title={rowData} onPress={()=>this.onSelectSize(rowData)}/>);
        break;
      default:break;
    }
  }

  renderSelectedValues(){
    return(
      <View style={styles.containerSelected}>
        <View style={styles.container}>
          {this.state.drink.type &&
            <View style={styles.margin}>
              <Button
                title={this.state.drink.type}
                onPress={()=>this.onSelectType()}
                color='#c8dcf4'/>
            </View>}
          {this.state.drink.style &&
            <View style={styles.margin}>
             <Button
               title={this.state.drink.style}
               onPress={()=>this.onSelectStyle()}
               color='#c8dcf4'/>
            </View>}
          {this.state.drink.size &&
            <View style={styles.margin}>
              <Button
                title={this.state.drink.size}
                onPress={()=>this.onSelectSize()}
                color='#c8dcf4'/>
            </View>}
        </View>
      </View>
    );
  }

  renderOptions(){
    if(this.state.step>2){
      return <ExtrasListView/>;
    }
    else{
      return <OptionsListView
        renderRow={this.renderRowSelect.bind(this)}
        dataSource={this.state.optionsList}/>;
    }
  }

  render() {
    return (
      <View style={{flex: 1, padding: 10}}>
        <Text>Current Selection:</Text>
        {this.renderSelectedValues()}
        <Text>Choose your {this.state.step>STEP_SIZE?"extras":"drink options"}:</Text>
        {!this.state.isLoading && this.renderOptions()}
        {this.state.isLoading && <ActivityIndicator style={styles.spinner} size="large"/>}
        <View style={{flexDirection:'row', justifyContent:'space-between'}}>
          <View style={{flex : 1, marginRight:4}}>
            <Button title='<' onPress={()=>this.onBack()} />
          </View>
          <View style={{flex : 8, marginLeft:4}}>
            {this._initialDrink &&
              <Options onSaveTitle='Update Drink' onSave={this.updateDrink.bind(this)}
                onSaveDisabled={!(this.state.step>STEP_SIZE) ||
                  (this.state.drink.style===this._initialDrink.style &&
                    this.state.drink.size===this._initialDrink.size)}/>}
            {!this._initialDrink &&
              <Options onSaveTitle='Add Drink' onSave={this.createDrink.bind(this)}
                onSaveDisabled={!(this.state.step>STEP_SIZE)}/>}
          </View>
        </View>
      </View>
    );
  }



  /*
  * API CALLS
  * TODO: PATH SHOULD CHANGE DEPENDING ON DRINK (IF THERE ARE DIFF DRINKS)
  */
  createDrink(){
    client.createDrink(this.state.order.id,this.state.drink)
    .then((responseJson) => {
      var newDrinks = this.state.order.coffees;
      if(!newDrinks){
        newDrinks = [responseJson];
      }
      else{
        newDrinks = [...newDrinks,responseJson];
      }
      var newOrder = this.state.order;
      newOrder.coffees = newDrinks;
      this.props.onComplete(this.state.order);
    })
  }

  updateDrink(){
    client.updateDrink(this.state.order.id,this.state.drink)
    .then((responseJson) => {
      this.props.onComplete(this.state.order);
    })
  }

  fetchDrinkOptions(drink){
    client.getDrinkMenu(drink)
    .then((responseJson) => {
      this.setState({
        drinkOptions:responseJson,
        isLoading:false
      });
      this.onStep(this.state.drink);
    })
  }
}

const styles = StyleSheet.create({
  spinner: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 80,
    flex: 1,
    padding: 8,
  },
  containerSelected:{
    flex: 0.45,
    alignItems: 'flex-start',
    alignSelf:'stretch',
  },
  container:{
    flex:1,
    alignItems:'stretch',
    alignSelf:'stretch',
  },
  margin:{
    marginTop: 8,
  }
});

export default Drink;
