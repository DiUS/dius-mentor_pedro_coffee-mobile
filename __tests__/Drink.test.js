import 'react-native';
import React from 'react';
import Drink from '../views/Drink.js';
import renderer from 'react-test-renderer';
import 'isomorphic-fetch';

var order = {
  id:1,
  coffees:null,
  name:'test_order',
  path:'order/1'
}

var drink = {
  id:1,
  style:'Latte',
  size:'3/4',
  path:'/order/1/coffee/1'
}

it('renders correctly', () => {
  const instance = renderer.create(
    <Drink order={order} drink={drink}/>
  );
  expect(instance.toJSON()).toMatchSnapshot();
});
