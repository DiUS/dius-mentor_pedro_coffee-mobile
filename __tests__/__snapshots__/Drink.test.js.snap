// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Current Selection:
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flex": 0.45,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#c8dcf4",
                  },
                ]
              }
            >
              Coffee
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#c8dcf4",
                  },
                ]
              }
            >
              Latte
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#c8dcf4",
                  },
                ]
              }
            >
              3/4
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Choose your 
    extras
    :
  </Text>
  <View>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 7,
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      renderSeparator={[Function]}
    >
      <View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Milk: Full-Cream
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              strength: Regular
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              sweetener: No
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              temperature: Regular
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              flavour: No
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              froth: Regular
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              others
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": 4,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            &lt;
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 8,
          "marginLeft": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 4,
              "marginLeft": 4,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
                "disabled",
              ]
            }
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#0C42FD",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#79d279",
                    },
                    Object {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Add Drink
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
